src/api/libcellml/entity.h

```diff
@@ -25,17 +27,100 @@ namespace libcellml {
  * @brief The Entity class.
  * Base class for all serialisable libCellML classes.
  */
-class Entity
+class LIBCELLML_EXPORT Entity
 {
-protected:
+public:
+    Entity(); /**< Constructor */
+    ~Entity(); /**< Destructor */
+    Entity(const Entity& rhs); /**< Copy constructor */
+    Entity(Entity &&rhs); /**< Move constructor */
+    Entity& operator=(Entity e); /**< Assignment operator */
+
+    /**
+     * @brief Serialise entity to std::string.
+     * @param format The format to serialise the object to.
+     * @return std::string representation of the object.
+     */
+    std::string serialise(libcellml::CELLML_FORMATS format) const
+    {
+        return doSerialisation(format);
+    }
```

Comment:
I would prefer the function body to be in a corresponding ```.cpp``` file. The same goes with ```getParent()```, ```setParent()```, etc. below.